File: articles/linear-algebra-for-systems-modeling/articles/gaussian-elimination-and-row-reduction/calculators/r/row_reduction_calculator.R
result <- data.frame(
  calculator = "row_reduction_rank_consistency_calculator",
  equation_count = 3,
  unknown_count = 3,
  pivot_columns = "0,1,2",
  coefficient_rank = 3,
  augmented_rank = 3,
  consistent = TRUE,
  solution_behavior = "unique solution",
  tolerance = 1.0e-10,
  warning = "Rank and solution behavior depend on tolerance and modeling assumptions."
)
dir.create("outputs", recursive = TRUE, showWarnings = FALSE)
write.csv(result, "outputs/r_row_reduction_calculator.csv", row.names = FALSE)
print(result)

File: articles/linear-algebra-for-systems-modeling/articles/gaussian-elimination-and-row-reduction/calculators/python/row_reduction_calculator.py
from __future__ import annotations
import csv, json
from pathlib import Path

def main() -> None:
    output_dir = Path("outputs")
    output_dir.mkdir(parents=True, exist_ok=True)
    result = {
        "calculator": "row_reduction_rank_consistency_calculator",
        "equation_count": 3,
        "unknown_count": 3,
        "pivot_columns": "0,1,2",
        "coefficient_rank": 3,
        "augmented_rank": 3,
        "consistent": True,
        "solution_behavior": "unique solution",
        "tolerance": 1.0e-10,
        "warning": "Rank and solution behavior depend on tolerance and modeling assumptions."
    }
    (output_dir / "row_reduction_calculator.json").write_text(json.dumps(result, indent=2, sort_keys=True), encoding="utf-8")
    with (output_dir / "row_reduction_calculator.csv").open("w", newline="", encoding="utf-8") as handle:
        writer = csv.DictWriter(handle, fieldnames=list(result.keys()))
        writer.writeheader()
        writer.writerow(result)
    print(json.dumps(result, indent=2, sort_keys=True))

if __name__ == "__main__":
    main()
